include/boost/gil/algorithm.hpp

```diff
@@ -189,7 +189,7 @@ namespace detail {
 /// \ingroup CopyPixels
 template <typename I, typename O>
 struct copier_n {
-    BOOST_FORCEINLINE void operator()(I src, typename std::iterator_traits<I>::difference_type n, O dst) const { std::copy(src,src+n, dst); }
+    BOOST_FORCEINLINE void operator()(I src, typename std::iterator_traits<I>::difference_type n, O dst) const { BOOST_ASSERT(n >= 0);  std::copy(src, src + n, dst); }
```

Comment:
The use of assertions is fine, I think, since it's `detail`, so called internally.

I just still remember discussions about using assertions on arguments of public functions:
- https://github.com/boostorg/gil/issues/360

Reply:
> I believe that one should never fail an assert on a runtime condition, like e.g. zero-sized file, file reading error, disk full, or such.

I agreee with this policy. In our case, the doc says for `std::copy_n`:

> Zero assignments if count < 0; count assignments otherwise.

so we should not assert but return early. Event if having `count < 0` is probably a programming mistake. I'll fix that.